File: backend/app/schemas/query.py
"""Pydantic models for Phase 3 NL query APIs."""

from __future__ import annotations

import uuid
from typing import Any

from pydantic import BaseModel, Field


class ClarificationItem(BaseModel):
    prompt_id: str
    choice: str


class NaturalLanguageRequest(BaseModel):
    question: str = Field(..., min_length=1)
    org_id: uuid.UUID | None = None
    disambiguation_token: str | None = None
    clarifications: list[ClarificationItem] | None = None


class SemanticVersionResponse(BaseModel):
    version_id: uuid.UUID
    version_label: str
    source_identifier: str | None
    content_sha256: str | None
    effective_from: str
    is_active: bool


class AuditListItem(BaseModel):
    log_id: uuid.UUID
    user_id: uuid.UUID | None
    natural_query: str
    status: str | None
    row_count: int | None
    execution_ms: int | None
    created_at: str
    correlation_id: uuid.UUID | None


class AuditListResponse(BaseModel):
    items: list[AuditListItem]
    next_cursor: str | None


class AuditDetailResponse(BaseModel):
    log_id: uuid.UUID
    tenant_id: uuid.UUID
    user_id: uuid.UUID | None
    correlation_id: uuid.UUID | None
    nl_session_id: uuid.UUID | None
    semantic_version_id: uuid.UUID | None
    natural_query: str
    resolved_plan: dict[str, Any] | None
    execution_ms: int | None
    row_count: int | None
    status: str | None
    error_message: str | None
    created_at: str

File: frontend/src/types/query.ts
/** Phase 3 NL query API shapes (aligned with api-contracts §8). */

export interface ClarificationChoice {
  id: string;
  label: string;
}

export interface ClarificationQuestion {
  prompt_id: string;
  text: string;
  choices: ClarificationChoice[];
}

export interface NaturalLanguageRequest {
  question: string;
  org_id?: string;
  disambiguation_token?: string;
  clarifications?: Array<{ prompt_id: string; choice: string }>;
}

export interface NaturalLanguageResponseCompleted {
  query_id: string;
  status: "completed";
  interpretation: string;
  columns: string[];
  rows: Record<string, string | null | undefined>[];
  disambiguation: null;
  semantic_version_label: string;
}

export interface NaturalLanguageResponseClarify {
  query_id: string;
  status: "needs_clarification";
  questions: ClarificationQuestion[];
  disambiguation: { token: string };
  semantic_version_label?: string;
}

export type NaturalLanguageResponse =
  | NaturalLanguageResponseCompleted
  | NaturalLanguageResponseClarify;

export interface AuditListItem {
  log_id: string;
  user_id: string | null;
  natural_query: string;
  status: string | null;
  row_count: number | null;
  execution_ms: number | null;
  created_at: string;
  correlation_id: string | null;
}

export interface AuditListResponse {
  items: AuditListItem[];
  next_cursor: string | null;
}
